### PR TITLE
cut-the-crap: init at 1.0.0

### PR DESCRIPTION
**Pull request by jappeace**
###### Motivation for this change

Adds a useful automated video editing tool for streamers.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

I think I'm the maintainer of this package

cc @

### CODE REVIEW

**Comment by cdepillabout** [comment]
@jappeace Would you consider releasing this to Hackage?

That way, it would get updates automatically here in nixpkgs whenever you released a new version.

**Comment by jappeace** [comment]
Hi @cdepillabout I uploaded it to [hackage](http://hackage.haskell.org/package/cut-the-crap-1.0.0).
It still needs ffmpeg as a runtime dependency, how does one add that?

**Comment by cdepillabout** [comment]
@jappeace Thanks!

Next time the hackage packages are updated on the `haskell-updates` branch, you should be able to see `cut-the-crap`.  You'll be able to try to build it with a command like `nix-build -A haskellPackages.cut-the-crap`.

> It still needs ffmpeg as a runtime dependency, how does one add that?

You should be able to add that with a wrapper script.  Here's an example:

https://github.com/NixOS/nixpkgs/blob/a8144b1ad33547cfdc31a903f8f7c0c658d78216/pkgs/development/haskell-modules/configuration-nix.nix#L585-L593

For cut-the-crap, it might look something like this:

```nix
  cut-the-crap =
    let path = stdenv.lib.makeBinPath [ ffmpeg ];
    in overrideCabal (addBuildTool super.futhark makeWrapper) (_drv: {
      postInstall = ''
        wrapProgram $out/bin/cut-the-crap \
          --prefix PATH : "${path}"
      '';
    });

File: pkgs/applications/video/cut-the-crap/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    optparse-generic regex-tdfa shelly system-filepath temporary text
+    unliftio-core
+  ];
+  testHaskellDepends = [
+    base exceptions generic-lens hspec hspec-core lens
+    optparse-applicative optparse-generic regex-tdfa shelly
+    system-filepath temporary text unliftio-core
+  ];
+  prePatch = "hpack";
+  license = stdenv.lib.licenses.mit;
```
Comment (doronbehar) [inline pkgs/applications/video/cut-the-crap/default.nix]: Why is there no `meta` as with any regular derivation? Maybe I'm just not familiar with the Haskell ecosystem...

File: pkgs/applications/video/cut-the-crap/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+, regex-tdfa, shelly, stdenv, system-filepath, temporary, text
+, unliftio-core, ffmpeg, makeWrapper
+}:
+(mkDerivation {
+  pname = "cut-the-crap";
+  version = "1.0.0";
+  src = fetchgit {
+    url = "https://github.com/jappeace/cut-the-crap";
+    sha256 = "1z4s0zan421q4722k32p0n33bsh7di1m6pbbdzqi7j87lfd9i7nk";
+    rev = "0ea8a0a0fd3e8d85f44337f13ef2445d22d1bf97";
```
Comment (doronbehar) [inline pkgs/applications/video/cut-the-crap/default.nix]: Why not use https://github.com/jappeace/cut-the-crap/releases/tag/1.0.0 ?

Comment (jappeace) [reply pkgs/applications/video/cut-the-crap/default.nix]: I made this before that debian thing. I wouldn't know how to use tags, or do you just mean the same commit as that tag?

File: pkgs/applications/video/cut-the-crap/default.nix (hunk 1 above)
Comment (jappeace) [reply pkgs/applications/video/cut-the-crap/default.nix]: This is generated, but at the moment I'm waiting for haskellPackages to be updated so I can just do cdepillabout's suggestion.

File: pkgs/applications/video/cut-the-crap/default.nix (hunk 2 above)
Comment (doronbehar) [reply pkgs/applications/video/cut-the-crap/default.nix]: Maybe that's also automatically generated...

**Comment by turion** [comment]
@cdepillabout didn't you mean _stackage_, not hackage?

**Comment by cdepillabout** [comment]
@jappeace What are you referring to?

`cut-the-crap` was originally not uploaded to Hackage, so it didn't automatically appear in the Haskell package set here in nixpkgs.

However, it appears that @jappeace has since uploaded to Hackage, so now it should be easy to just use `haskellPackages.cut-the-crap`.

Although you're right that uploading to Stackage (in addition to Hackage) would make it easier to make sure `cut-the-crap` builds here in nixpkgs.

**Comment by turion** [comment]
@cdepillabout Thanks for the clarification! I can never remember how, in which files, and in which workflow either hackage or stackage are crawled. (Wish this was documented.)

**Comment by turion** [comment]
@cdepillabout Thanks for the clarification! I can never remember how, in which files, and in which workflow either hackage or stackage are crawled. (Wish this was documented.)

**Comment by cdepillabout** [comment]
@turion Yeah, I definitely sympathize!

A (very short) explanation is that Haskell packages are updated from Hackage automatically on a daily basis in this file:

https://github.com/NixOS/nixpkgs/blob/16477d764f375ef7b81acbfd13189299ff67f233/pkgs/development/haskell-modules/hackage-packages.nix

The packages in the stackage resolver are specified here:

https://github.com/NixOS/nixpkgs/blob/16477d764f375ef7b81acbfd13189299ff67f233/pkgs/development/haskell-modules/configuration-hackage2nix.yaml#L45-L80

This list is updated once a week or so by peti.  I believe he does it semi-automatically.

That file also has a bunch of other settings for special things to do when updating packages from Hackage.

**Comment by jappeace** [comment]
I'll fix this up right now as discussed above

**Comment by jappeace** [comment]
Ok nice, that's a lot less code and it makes updating this as simple as re-uploading to hackage. Thanks for all the help @cdepillabout

**Comment by cdepillabout** [comment]
@jappeace One more thing.  Could you rebase this on the `haskell-updates` branch, and change the base branch for this PR here on GitHub to `haskell-updates`?

We like to have most Haskell-related changes going to the `haskell-updates` branch.

**Comment by jappeace** [comment]
lol, that upsets github quite a bit @cdepillabout

**Comment by jappeace** [comment]
I don't want to deal with github bullshit, I kindly refer you to https://github.com/NixOS/nixpkgs/pull/85438

Sorry for everyone else who got tagged. I didn't expect github to be so un-intelligent.

**Comment by doronbehar** [comment]
@jappeace please mind your language :pray: . It's 1 thing we add here a program that has "crap" in it's name - it doesn't mean we have to use such words when we deal with them.

**Comment by jappeace** [comment]
@doronbehar I already self censored. You probably had a tab open or something.

**Comment by doronbehar** [comment]
I noticed that and I still felt it was necessary to put it. I was subscribed to this issue and I got that by email.

**Pull request closed without merging**